scipy/_lib/_array_api.py

```diff
@@ -44,7 +44,7 @@
     ArrayLike = Array | npt.ArrayLike
 
 
-def compliance_scipy(arrays: list[ArrayLike]) -> list[Array]:
+def compliance_scipy(arrays: list[ArrayLike], sparse_ok: bool = True) -> list[Array]:
```

Comment:
If we want to add this keyword, then surely the default should be `False`? Sparse arrays are accepted very rarely outside of `scipy.sparse` itself, and they don't have a compatible API with numpy arrays or the array API standard.

Reply:
I think I see why you've added it at all - what is missing is `array_namespace` rejecting sparse arrays. Once that is added, this keyword isn't needed. The one call you're trying to replace is:
```python
a = _asarray_validated(a, check_finite=False)
```
which means it should be translatable to a plain `array_namespace(a)` call once we plug this strictness hole.

Reply:
Cool, I'm certainly happy with a default of `False` and even not having the keyword at all - I was trying to leave this open for potentially more functions accepting `scipy.sparse` arrays in the future, but I suppose we can cross that bridge when we come to it. I went with the default not making `array_namespace` more strict for now to be minimal, but happy to do so.

Reply:
Yes, let's do that - that will be an improvement. Right now if you feed sparse array/matrix instances to most functions, you can get weird exceptions or incorrect results (calling `np.asarray` on a sparse array results in a size- object array ...).

Reply:
I checked and nowhere in the codebase do we use `sparse_ok=True` - so yes, I'll remove the keyword.